### PR TITLE
Switch GCP jobs to push images to api.ci for PRs, then use them during the cluster deploy

### PR DESCRIPTION
**Pull request by smarterclayton**
@stevekuznetsov I'm probably going to have to debug this in production

### CODE REVIEW

File: sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce.yml [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 overrides:
-  junit_analysis: False
-  sync_repos:
-    - name: "openshift-ansible"
-      type: "pull_request"
-    - name: "origin"
-    - name: "release"
-    - name: "image-registry"
-    - name: "kubernetes-metrics-server"
+  sync:
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce.yml]: you need `openshift,release` in these

File: sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce_37.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+parent: 'common/test_cases/origin_release_install_gce.yml'
+overrides:
+  sync:
+    - "openshift,origin=master"
+    - "openshift,aos-cd-jobs=master"
+    - "openshift,image-registry=master"
+    - "openshift,kubernetes-metrics-server=master"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce_37.yml]: master for 3.7?

Comment (smarterclayton) [reply sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce_37.yml]: i'll just remove them - 3.7 doesn't use these at all.

Comment (smarterclayton) [reply sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce_37.yml]: Actually I need to sync something or the job will fail.

Comment (smarterclayton) [reply sjb/config/test_cases/test_pull_request_openshift_ansible_extended_conformance_gce_37.yml]: These repos wouldn't build before 3.9

**Comment by smarterclayton** [comment]
This is working except for openshift/openshift-ansible#7502

**Pull request merged**